internal/reflect/into.go

```diff
@@ -148,6 +148,14 @@ func BuildValue(ctx context.Context, typ attr.Type, val tftypes.Value, target re
 
 		return target, diags
 	}
+
+	// If the attr.Type has a dynamic value, then we need to use the tftypes.Value
+	// to determine the attr.Type for the reflection logic below.
+	dynTyp, ok := typ.(attr.TypeWithDynamicValue)
+	if ok {
+		typ = dynTyp.DetermineAttrType(val.Type())
+	}
```

Comment:
As of right now, this is the main reason why I had to utilize the `attr` package with `attr.TypeWithDynamicValue`, rather than the dynamic valuable interfaces.

I also use it in the `(Attribute).ValidateImplementation`, but those may be able to use the valuable interfaces

Reply:
This has been removed and the proper error message is being returned in it's place. See https://github.com/hashicorp/terraform-plugin-framework/pull/931#discussion_r1528442316